preprocessing/generate_flickr30k_inference_data.py

```diff
@@ -0,0 +1,143 @@
+      tf.data.TFRecordDataset,
+      cycle_length=8,
+      num_parallel_calls=tf.data.experimental.AUTOTUNE)
+  dataset = dataset.map(
+      parse_fn,
+      num_parallel_calls=tf.data.experimental.AUTOTUNE)
+
+  image_key_to_index = dict()
+  text_key_to_index = dict()
+  num_img_per_shard = TOPK_IMAGES // NUM_SHARDS
+  num_txt_per_shard = max_num_examples[split]['text'] // NUM_SHARDS
+
+  ii = 0
+  for i, ex in enumerate(dataset, start=1):
+    image_key = ex['image/key'].numpy()
+
+    if len(image_key_to_index) < TOPK_IMAGES:
+
+      image_index = image_key_to_index.get(image_key, len(image_key_to_index))
+      image_key_to_index[image_key] = image_index
+      img_int_features = {'image_index': image_index}
+      img_string_features = {'image_key': image_key}
+      tf_ex = utils.image_example(ex['image/encoded'], 
+                                  img_string_features,
+                                  img_int_features)
+      tf_ex_string = tf_ex.SerializeToString()
+
+      is_last_shards = img_shard_idx == NUM_SHARDS - 1
+      if (len(image_key_to_index) % num_img_per_shard == 0 and 
+              not is_last_shards):
+        img_writer.close()
+        img_shard_idx += 1
+        tfrecord_filename = TFRECORD_BASENAME.format(
+            EVAL_DATA_DIR, split, 'image', img_shard_idx, NUM_SHARDS)
+        img_writer = tf.io.TFRecordWriter(tfrecord_filename)
+      img_writer.write(tf_ex_string)
+
+    # Text
+    # 5 captions per image
+    captions = ex['caption/tokenized_text'].numpy()
+    for idx, caption in enumerate(captions):
+      text_key = f'{image_key.decode("utf-8")}_{idx}'.encode()
+      text_index = text_key_to_index.get(text_key, len(text_key_to_index))
+      text_key_to_index[text_key] = text_index
+      gt_image_index = image_key_to_index.get(image_key, -1)
+      txt_string_features = {
+          'caption': caption,
+          'text_key': text_key,
+      }
+      txt_int_features = {
+          'text_index': text_index,
+          'gt_image_index': gt_image_index,
+      }
+      tf_ex = utils.text_example(txt_string_features, txt_int_features)
+      tf_ex_string = tf_ex.SerializeToString()
+
+      is_last_shards = txt_shard_idx == NUM_SHARDS - 1
+      if (len(text_key_to_index) % num_txt_per_shard == 0 and 
+              not is_last_shards):
+        txt_writer.close()
+        txt_shard_idx += 1
+        tfrecord_filename = TFRECORD_BASENAME.format(
+            EVAL_DATA_DIR, split, 'text', txt_shard_idx, NUM_SHARDS)
+        txt_writer = tf.io.TFRecordWriter(tfrecord_filename)
+      txt_writer.write(tf_ex_string)
+
+    if i % 100 == 0:
+      print(f'  Process {i}')
+
+  split_input_meta_data = {
+      f"{split}_image_input_path": f'{EVAL_DATA_DIR}/flickr30k.{split}.image.recordio-*',
+      f"{split}_text_input_path": f'{EVAL_DATA_DIR}/flickr30k.{split}.text.recordio-*',
+      f"{split}_num_image_examples": len(image_key_to_index),
+      f"{split}_num_text_examples": len(text_key_to_index),
```

Comment:
Nit: strings should be `'` quoted instead of `"` to be consistent with the rest of the file.